src/Symfony/Component/Messenger/Transport/SqsExt/Connection.php

```diff
@@ -0,0 +1,362 @@
+        }
+
+        return 0;
+    }
+
+    public function requeue(string $id, int $delay = 0): void
+    {
+        $this->call($this->getQueueUrl(), [
+            'Action' => 'ChangeMessageVisibility',
+            'ReceiptHandle' => $id,
+            'VisibilityTimeout' => $delay,
+        ]);
+    }
+
+    public function requeuePrefetchedMessages(): void
+    {
+        if (null !== $this->currentResponse) {
+            $this->currentResponse->cancel();
+        }
+
+        foreach ($this->getPendingMessages() as $message) {
+            $this->requeue($message['id']);
+        }
+    }
+
+    public function send(string $body, array $headers, int $delay = 0): void
+    {
+        if ($this->configuration['auto_setup']) {
+            $this->setup();
+        }
+
+        $this->call($this->getQueueUrl(), [
+            'Action' => 'SendMessage',
+            'MessageBody' => json_encode(['body' => $body, 'headers' => $headers]),
+            'DelaySeconds' => $delay,
+        ]);
+    }
+
+    private function getQueueUrl(): string
+    {
+        if (!isset($this->queueUrl)) {
+            $response = $this->request($this->configuration['endpoint'], [
+                'Action' => 'GetQueueUrl',
+                'QueueName' => $this->configuration['queue_name'],
+            ]);
+            $this->checkResponse($response);
+            $xml = new \SimpleXMLElement($response->getContent());
+
+            $this->queueUrl = (string) $xml->GetQueueUrlResult->QueueUrl;
+        }
+
+        return $this->queueUrl;
+    }
+
+    private function call(string $endpoint, array $body): void
+    {
+        $this->checkResponse($this->request($endpoint, $body));
+    }
+
+    private function request(string $endpoint, array $body): ResponseInterface
+    {
+        if (!$this->configuration['access_key']) {
+            return $this->client->request('POST', $endpoint, ['body' => $body]);
+        }
+
+        $region = $this->configuration['region'];
+        $service = 'sqs';
+
+        $method = 'POST';
+        $requestParameters = http_build_query($body, '', '&', PHP_QUERY_RFC1738);
+        $amzDate = gmdate('Ymd\THis\Z');
+        $parsedUrl = parse_url($endpoint);
+
+        $headers = [
+            'host' => $parsedUrl['host'],
+            'x-amz-date' => $amzDate,
+            'content-type' => 'application/x-www-form-urlencoded',
+        ];
+
+        $signedHeaders = ['host', 'x-amz-date'];
+        $canonicalHeaders = implode("\n", array_map(function ($headerName) use ($headers): string {
+            return sprintf('%s:%s', $headerName, $headers[$headerName]);
+        }, $signedHeaders))."\n";
+
+        $canonicalRequest = implode("\n", [
+            $method,
+            $parsedUrl['path'] ?? '/',
+            '',
+            $canonicalHeaders,
+            implode(';', $signedHeaders),
+            hash('sha256', $requestParameters),
+        ]);
+
+        $algorithm = 'AWS4-HMAC-SHA256';
```

Comment:
What about using constant for that ?

Reply:
that's a detail of implementation of the private method `request`. I'd probably use a constant if this class was `AwsRequestSignerV4`